ast/src/parsed/display.rs

```diff
@@ -553,6 +557,68 @@ impl Display for FunctionDefinition {
     }
 }
 
+impl Display for TraitDeclaration<Expression> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let type_vars = if self.type_vars.is_empty() {
+            Default::default()
+        } else {
+            format!("<{}>", self.type_vars)
+        };
+        write!(
+            f,
+            "trait {name} {type_vars} {{\n{methods}}}",
+            name = self.name,
+            methods = indent(self.methods.iter().map(|m| format!("{m},\n")).format(""), 1)
+        )
+    }
+}
+
+impl Display for TraitMethod<Expression> {
```

Comment:
I think since we do not have an implicit Self type, this should be TraitFunction instead.